src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,79 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  id: string;
+}
+
+class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_RESTORE;
+  }
+
+  public get description(): string {
+    return 'Restores given items from the site recycle bin';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Restoring items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
```

Comment:
We should escape user input before passing it to the command

Reply:
@waldekmastykarz I am like thinking is this really necessary as the siteUrl input is validated after all 🤔. What do you think?

Reply:
Currently we only check if it starts with `https` so theoretically someone could still abuse it

Reply:
if I escape siteUrl (for example "https://something.com" then I get "https%3A//something.com" which basically wont work) I checked and also in similar commands like spo list list where we also pass WebUrl we also don't escape this parameter and maybe for the same reason 🤔

Reply:
My bad. Good of you to check with other commands. Let's keep it as-is